### PR TITLE
Add initial testing framework

### PR DESCRIPTION
**Pull request by Seth10001**

### CODE REVIEW

File: .github/workflows/unit-test.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+name: C++ Unit Tests
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
```
Comment (John-Carr) [inline .github/workflows/unit-test.yml]: Wana freeze this to something like 20.04 (latest LTS?)

File: .gitmodules [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[submodule "cpputest"]
+	path = cpputest
```
Comment (John-Carr) [inline .gitmodules]: Should we put this under tests/cpputest? That way all the unit test stuff is contained

**Review by John-Carr — APPROVED** [review]

File: .gitmodules (hunk 2 above)
Comment (Seth10001) [reply .gitmodules]: Good idea! Went ahead and did that in fd4005d5525a27cdfe3efb85244238a9ff51bfe1

File: .github/workflows/unit-test.yml (hunk 1 above)
Comment (Seth10001) [reply .github/workflows/unit-test.yml]: Yep, I did that in 242990419ed3f3607aa9c7246f99efa2cd303c36

**Pull request merged**